src/main/java/seedu/lifeasier/storage/FileStorage.java

```diff
@@ -0,0 +1,151 @@
+    private Ui ui;
+    private NoteStorage noteStorage;
+    private TaskStorage taskStorage;
+
+    public FileStorage(String fileNameTasks, String fileNameNotes, Ui ui, NoteList notes, TaskList tasks) {
+        this.filePathTasks = DIRECTORY_PATH + fileNameTasks;
+        this.filePathNotes = DIRECTORY_PATH + fileNameNotes;
+        this.ui = ui;
+        this.noteStorage = new NoteStorage(notes, filePathNotes);
+        this.taskStorage = new TaskStorage(tasks, filePathTasks);
+    }
+
+    /**
+     * Reads and loads the stored information on the save file. If save files are missing, the method will create a new
+     * save file and directory.
+     */
+    public void readSaveFiles() {
+        File saveFileTasks = new File(filePathTasks);
+        File saveFileNotes = new File(filePathNotes);
+
+        if (!fileExists(saveFileTasks, saveFileNotes)) {
+            createNewSaves(saveFileTasks, saveFileNotes);
+        } else {
+            System.out.println("Reading your save data...");
+            noteStorage.readNotesSave(filePathNotes);
+            taskStorage.readTasksSave(filePathTasks);
+            System.out.println("All save data successfully read!");
+        }
+    }
+
+    /**
+     * Checks if both save files exist.
+     *
+     * @param saveFileTasks File object with file path to the save file which contains saved task information.
+     * @param saveFileNotes File object with file path to the save file which contains saved note information.
+     * @return True when both files exist.
+     */
+    private boolean fileExists(File saveFileTasks, File saveFileNotes) {
+        return saveFileTasks.exists() && saveFileNotes.exists();
+    }
+
+    /**
+     * Determines if save directory and files are existent, before creating new save files and directories.
+     */
+    private void createNewSaves(File saveFileTasks, File saveFileNotes) {
+        File fileDirectory = new File(DIRECTORY_PATH);
+
+        //If directory exists, create missing save files
+        if (directoryExists(fileDirectory)) {
+            ui.showDirectoryDetected();
+
+            if (!saveFileTasks.exists()) {
+                createTaskSaveFile(saveFileTasks);
+            }
+
+            if (!saveFileNotes.exists()) {
+                createNoteSaveFile(saveFileNotes);
+            }
+
+        } else {
+            handleMissingSaveDirectory(fileDirectory, saveFileTasks, saveFileNotes);
+        }
+    }
+
+    private boolean directoryExists(File fileDirectory) {
+        return fileDirectory.exists();
+    }
+
+    /**
+     * Handles when no save directory is detected and creates a new directory.
+     *
+     * @param fileDirectory File object which holds the directory path to be created.
+     */
+    private void handleMissingSaveDirectory(File fileDirectory, File saveFileTasks, File saveFileNotes) {
+        ui.showNoDirectoryDetected();
+        //Attempt creation of new save directory to hold save files
```

Comment:
Similar to a previous comment, the JavaDoc comment and good variable naming makes this comment seem unnecessary.